src/gyroaverage/gyroaverage_operator.hpp

```diff
@@ -0,0 +1,213 @@
+            CoordinateTransformFunction coordinate_transform,
+            std::size_t nb_gyro_points = 8) const
+    {
+        static_assert(
+                std::is_invocable_v<CoordinateTransformFunction, double, double>,
+                "CoordinateTransformFunction must be a functor on (double, double)");
+        IdxRangeR const r_domain = get_idx_range<GridR>(A);
+        IdxRangeTheta const theta_domain = get_idx_range<GridTheta>(A);
+        IdxRangeBatch const batch_domain = get_idx_range<GridBatch>(A);
+        IdxRangeRTheta const rtheta_mesh = get_idx_range<GridR, GridTheta>(A);
+
+        ddc::NullExtrapolationRule r_extrapolation_rule;
+        ddc::PeriodicExtrapolationRule<DimensionThetaType> theta_extrapolation_rule;
+
+        SplineRThetaBuilder const spline_builder(rtheta_mesh);
+        SplineRThetaEvaluatorNullBound const spline_evaluator(
+                r_extrapolation_rule,
+                r_extrapolation_rule,
+                theta_extrapolation_rule,
+                theta_extrapolation_rule);
+
+        // Instantiate chunk of spline coefs to receive output of spline_builder (r, theta)
+        DFieldMemBSRTheta coef_alloc(
+                spline_builder.batched_spline_domain(rtheta_mesh),
+                ddc::DeviceAllocator<double>());
+        DFieldBSRTheta const coef = get_field(coef_alloc);
+        DConstFieldRTheta const rho_L = get_const_field(m_rho_L);
+
+        using SubDFieldRTheta = DField<
+                IdxRangeRTheta,
+                typename ExecutionSpace::memory_space,
+                Kokkos::layout_stride>;
+
+        ddc::for_each(batch_domain, [&](IdxBatch const ib) {
+            // FIXME
+            // The input of the spline builder must be LayoutRight
+            // We allocate a buffer in LayoutRight whereto the slice is copied
+            SubDFieldRTheta const sub_A = A[ib];
+            SubDFieldRTheta sub_A_bar = A_bar[ib];
+            DFieldMemRTheta sub_A_alloc(rtheta_mesh, ddc::DeviceAllocator<double>());
+            ddc::parallel_deepcopy(sub_A_alloc, sub_A);
+            spline_builder(coef, get_const_field(sub_A_alloc));
+
+            // Loop over r, theta
+            ddc::parallel_for_each(
+                    rtheta_mesh,
+                    KOKKOS_LAMBDA(IdxRTheta const irtheta) {
+                        IdxR const ir(irtheta);
+                        IdxTheta const itheta(irtheta);
+                        double const r = ddc::coordinate(ir);
+                        double const theta = ddc::coordinate(itheta);
+
+                        // Average over gyro points
+                        double tmp = 0.0;
+                        for (std::size_t igyro = 0; igyro < nb_gyro_points; igyro++) {
+                            // Compute the particle position in (R, Z) coordinate
+                            double const alpha = M_PI * 2.0 / static_cast<double>(nb_gyro_points)
+                                                 * static_cast<double>(igyro);
```

Comment:
It looks a mesh over $[0, 2\pi]$. I am not convinced it makes sense to use DDC for this but could it make sense to have a dedicated functor ?